src/util/run-as-root.js

```diff
@@ -72,23 +68,41 @@ const sudoPromptToRunShell = async cmd =>
  * from the current environment, pass it in through the `args` array and receive
  * it as a parameter.
  *
- * @param {Function|string} fn JavaScript code to run. Must be a function or a
- *     string that evaluates to a function. It can take arguments, which must be
- *     passed in order in an array to the following `args` parameter.
+ * @param {String} prompt Custom prompt to display. Optional, but highly
+ *     recommended. [sudo -p](https://www.sudo.ws/man/1.8.17/sudo.man.html#p)
+ *     variables are interpolated from this string.
+ * @param {Function} fn JavaScript code to run. Must be a function. It can take
+ *     arguments, which must be passed in order in an array to the following
+ *     `args` parameter.
  * @param {Array} args An array of values to be passed as arguments. Must be
  *     serializable to JSON.
  * @returns {Promise<string>} A promise for the console output of the
  *     evaluated code. Rejects if the user did not authorize, or if the code
  *     threw an exception.
  */
 module.exports = async (fn, ...args) => {
```

Comment:
I see what's going on here with `prompt`, `fn`, and `func`, but it's a bit difficult to read. Maybe that's unavoidable.

It might be clearer to call argument 0 `prompt` instead of `fn`, default it to `'Temporary...'`, and go from there—you might be able to get rid of either `fn` or `func` that way.

Reply:
OK! I'm happy to make the change. I almost think that a description should simply be required, which would be more readable and maybe safer, since this is not a beginner-facing tool.

Reply:
Great, thanks for making that change. 👍